### ISSUE TITLE
HPC-GAP: copying atomic lists and records is broken

### ISSUE THREAD

**Issue by fingolfin** [gap-system/gap#1741]
Copying of atomic lists and records is fundamentally broken: this operation returns either the atomic object itself, or some weird error. To demonstrate this:
```
gap> foo:=AtomicList([]);
<atomic list of size 0>
gap> IsMutable(foo);   # atomic objects are always mutable
true
gap> bar:=StructuralCopy(foo);
<atomic list of size 0>
gap> IsIdenticalObj(foo,bar);   # BUG: copy of a mutable object must not equal the original
true
gap> qux:=Immutable(foo);
<atomic list of size 0>
gap> IsMutable(qux);     # BUG: result of Immutable should not be mutable
true
gap> IsIdenticalObj(foo,qux);   # ... and not identical to original
true
gap> bar:=ShallowCopy(foo);   # BUG: this should give a less paniced error -- or just work
Error, Panic: tried to shallow copy object of unsupported type 'atomic list'
```
One way to fix this would be to simply disallow copying of atomics.

We could also try to implement it... Making an immutable copy then would just return an immutable plist or precord. And a mutable copy a new atomic object identical subobjects -- at least if the original atomic is (strict) write once. We might want to disallow copies for read/write atomics, for simplicity?

@rbehrends Any thoughts?

**Comment by ChrisJefferson** [comment]
I think we should forbid structuralcopy or shallowcopy of mutable atomic lists, for the same reason we should also forbid `=` and `<`: They are impossible to serialise (that is, you can end up with a copy which has a value the atomiclist never actually help, or compare two lists equal that at no point in time were ever equal).

The mutable I have less strong opinions on, but I'm tempted to forbid that too, mainly to try to teach people that `AtomicList` and `AtomicRecord` should not be treated like every other HPC-GAP object, and are easy to abuse.

**Comment by fingolfin** [comment]
This has been resolved by PR #1749
